xchange-core/src/main/java/org/knowm/xchange/service/marketdata/MarketDataService.java

```diff
@@ -177,4 +179,25 @@ default Trades getTrades(Instrument instrument, Object... args) throws IOExcepti
   default Trades getTrades(Params params) throws IOException {
     throw new NotYetImplementedForExchangeException("getTrades");
   }
+
+
+  /**
+   * Get the CandleStickData for given currency between startDate to endDate.
+   *
+   * @param currencyPair currencyPair.
+   * @param startDate startDate of candle stick, E.G. march 2022.
+   * @param endDate endDate of candle stick, E.G. april 2022.
+   * @param args extra exchange specific parameters.
+   * @return The CandleStickData, null if some sort of error occurred. Implementers should log the error.
+   * @throws ExchangeException - Indication that the exchange reported some kind of error with the
+   * request or response
+   * @throws NotAvailableFromExchangeException - Indication that the exchange does not support the
+   * requested function or data
+   * @throws NotYetImplementedForExchangeException - Indication that the exchange supports the
+   * requested function or data, but it has not yet been implemented
+   * @throws IOException - Indication that a networking error occurred while fetching JSON data
+   */
+  default CandleStickData getCandleStickData(CurrencyPair currencyPair, Date startDate, Date endDate, Object... args) throws IOException {
+    throw new NotYetImplementedForExchangeException("getCandleStickData");
+  }
```

Comment:
why must we add `CandleStick` when we have `Ticker` which has essentially the same fields?

why can't you use this existing interface method

```
List<Ticker> getTickers(Params params) throws IOException {
```

to implement this feature

Reply:
I was just going to ask the same thing. ;)

Reply:
At different generic exchange library implementations(like ccxt), ticker and candle data and methods are designed separately.

We should add a new method because exchange sites have separate services for tickers and candlesticks.

ticker and candle data are separate concepts. They may seem similar due to that they show information about a currency trading pair. But in fact, the candlestick is temporal information with many additional fields, such as high, low price, and open, close price. And also it has a definite time frame.

![image](https://user-images.githubusercontent.com/3147731/167272132-79b2602d-cc1a-4ccc-af2e-a1216e03bb8c.png)

Actually, we assume it has been modeled better in exchange, but we see that lots of concepts are missed or modeled incorrectly. So we did our best to model it appropriately.

Reply:
Any comments? @walec51 @timmolter

Reply:
https://www.investopedia.com/terms/s/stocksymbol.asp actually describes a concept more align with a Currency Symbol and not out ticker class

the concept of a candleStick is very similar as the "ticker" term in our code - the "ticket" could be described as the last 24h candle stick for a given currency pair

however I'm starting to see and acknowledge we need a separate model as the method I've suggested is usually used to get those tickers for multiple currency pairs and not for many intervals of one currency pair

so taking this to account adding a `CandleStick` model makes sense and in the future the `Ticker` model could look like this

```
Ticker {
  Instrument instrument;
  CandleStick lastCandleStick; 
  // ...
}
```

that being said I think you should try to keep names of fields your `CandleStick` class in the same convention as they are seen in `Ticker`

```
  private final BigDecimal open;
  private final BigDecimal last; // would be also useful to add
  private final BigDecimal high;
  private final BigDecimal low;
  private final BigDecimal volume;
  private final BigDecimal quoteVolume; // instead of currencyVolume
  private final BigDecimal vwap; // these 5 fields can be null if not provided by the exchange
  private final BigDecimal bid; 
  private final BigDecimal bidSize;
  private final BigDecimal ask;
  private final BigDecimal askSize;
  private final Date timestamp; // instead of date
```

Reply:
Added new fields and renamed the ones with the suggested alternatives.